### PR TITLE
change data transform language

### PR DESCRIPTION
**Pull request by blanchco**
# Description

* Ability to change programming languages in data transform
* fixed an issue where the ConfirmDialog was popping up multiple times when used, this should be defined at the root of the project

https://github.com/DARPA-ASKEM/terarium/assets/33158416/22e3141e-5491-4585-aad6-a4c447773eff

### CODE REVIEW

File: packages/client/hmi-client/src/components/dataset/tera-dataset-jupyter-panel.vue [hunk 1]
```diff
@@ -234,11 +245,49 @@ jupyterSession.kernelChanged.connect((_context, kernelInfo) => {
...
+					language: val.value,
+					context_info: contextInfo
+				},
+				msgType: 'context_setup_request',
+				msgId: createMessageId('context_setup_request')
+			};
+			const message: JupyterMessage = createMessage(messageBody);
+			kernel.sendJupyterMessage(message);
+
+			emit('update-language', val.value);
```
Comment (mwdchang) [inline packages/client/hmi-client/src/components/dataset/tera-dataset-jupyter-panel.vue:286]: Does ordering matter? I don't know if you have the guarantee that `kernel.sendJupyterMessage(message);` is finished at this stage.

Comment (blanchco) [reply packages/client/hmi-client/src/components/dataset/tera-dataset-jupyter-panel.vue:286]: You're right, I'm making a change to listen to the `context_setup_response` to get the language, that should give use the language change

Comment (blanchco) [reply packages/client/hmi-client/src/components/dataset/tera-dataset-jupyter-panel.vue]: @mwdchang it _should_ be good now with the latest commit

File: packages/client/hmi-client/src/types/common.ts [hunk 2]
```diff
@@ -212,3 +213,25 @@ export interface NotificationItem {
...
+ * @returns {Array} An array of options for programming languages.
+ */
+export const programmingLanguageOptions = (): { name: string; value: string }[] =>
+	Object.values(ProgrammingLanguage)
+		.filter((lang) => lang !== ProgrammingLanguage.Zip)
+		.map((lang) => ({
+			name:
+				lang && `${lang[0].toUpperCase() + lang.slice(1)} (${ProgrammingLanguageVersion[lang]})`,
+			value: ProgrammingLanguageVersion[lang]
+		}));
```
Comment (mwdchang) [inline packages/client/hmi-client/src/types/common.ts:237]: This feels like an awkward construct, wouldn't it be easier if we start off by storing the programming languages and their versions together? Then we don't need to do these kind of data wrangling and have the slight problem of independently maintaining "languages" across two physical locations in the repo.

Just food for thought for later - no need to actionnow.

**Review by mwdchang — APPROVED** [review]

File: packages/client/hmi-client/src/types/common.ts (hunk 2 above)
Comment (blanchco) [reply packages/client/hmi-client/src/types/common.ts:237]: might be good to use the beaker api to fetch the languages and version to use in the dropdown in the future: `https://beaker.terarium.ai/contexts`

**Pull request merged**